### PR TITLE
Oracle refactor work

### PR DESCRIPTION
**Pull request by Aryan-Verma**
Refactor work for Oracle plugin

### CODE REVIEW

File: database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java [hunk 1]
```diff
@@ -146,7 +146,7 @@ public Schema getSchema(Class<? extends Driver> driverClass) throws IllegalAcces
     }
   }
 
-  private Schema loadSchemaFromDB(Connection connection, String query) throws SQLException {
+  public Schema loadSchemaFromDB(Connection connection, String query) throws SQLException {
```
Comment (MrRahulSharma) [inline database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java]: Do not require this anymore.

File: database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java [hunk 2]
```diff
@@ -165,7 +165,7 @@ static String removeConditionsClause(String importQueryString) {
     return query;
   }
 
-  private Schema loadSchemaFromDB(Class<? extends Driver> driverClass)
+  public Schema loadSchemaFromDB(Class<? extends Driver> driverClass)
```
Comment (wyzhang) [inline database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java]: This method doesn't seem a good candidate to be public. The one above seems to be more appropriate for public. 

Can the method be simply loadSchemaFromDB(connection) ? What's is query string used for?

File: database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java [hunk 3]
```diff
@@ -229,22 +229,46 @@ public void prepareRun(BatchSourceContext context) throws Exception {
...
+    lineageRecorder.createExternalDataset(schema);
+    if (schema != null && schema.getFields() != null) {
+      lineageRecorder.recordRead("Read", "Read from database plugin",
+                                 schema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList()));
+    }
+    context.setInput(Input.of(sourceConfig.getReferenceName(), new SourceInputFormatProvider(
+      DataDrivenETLDBInputFormat.class, connectionConfigAccessor.getConfiguration())));
+  }
+
+  public ConnectionConfigAccessor getConnectionConfigAccessor (String driverClassName,
```
Comment (wyzhang) [inline database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java:285]: nit: Please add java doc for what this method is for and what it does

File: database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java (hunk 1 above)
Comment (wyzhang) [reply database-commons/src/main/java/io/cdap/plugin/db/source/AbstractDBSource.java]: If made public, please add java doc

File: database-commons/src/main/java/io/cdap/plugin/util/DriverCleanup.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
  * class to de-register driver
  */
 public class DriverCleanup implements Destroyable {
-  private final JDBCDriverShim driverShim;
+  public final JDBCDriverShim driverShim;
```
Comment (wyzhang) [inline database-commons/src/main/java/io/cdap/plugin/util/DriverCleanup.java]: Don't simply make this public. Use accessor at least.

**Review by saimukkamala — APPROVED** [review]

**Pull request merged**